src/eva/data/ioda_obs_space.py

```diff
@@ -0,0 +1,96 @@
+# (C) Copyright 2021-2022 NOAA/NWS/EMC
+#
+# (C) Copyright 2021-2022 United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration. All Rights Reserved.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+
+# --------------------------------------------------------------------------------------------------
+
+import xarray as xr
+
+from eva.eva_base import EvaBase
+from eva.utilities.config import get
+
+# --------------------------------------------------------------------------------------------------
+
+
+class IodaObsSpace(EvaBase):
+
+    def execute(self, data_collections):
+
+        # Local copies
+        logger = self.logger
+        config = self.config
+
+        # Get the list of datasets to create
+        datasets = config.get('datasets')
+
+        # Loop over the datasets
+        for dataset in datasets:
+
+            # Get the groups to be read
+            groups = get(dataset, logger, 'groups')
+
+            # Assert the correct use of variables and channels
+            if ('variables' in dataset and 'channels' in dataset):
+                self.logger.abort('Dataset configuraiton for \'' + dataset['name'] + '\'' +
+                                  'contains both channels or variables, should be one or ' +
+                                  'the other. If channels present, variable is ' +
+                                  'brightness_temperature')
+
+            # Get channels or variables
+            if ('variables' in dataset):
+                variables = dataset['variables']
+            elif ('channels' in dataset):
```

Comment:
This is probably okay for now, but there will eventually be cases where 'channels' could also refer to AOD, or antenna temperature, etc. and not just brightness_temperature.

Reply:
Ah yes good point. I'll set it to allow either variables or channels and variables. Is it safe to set it such that if channels are present the variables list must be length 1?